forge/forge/verify/value_checkers.py

```diff
@@ -0,0 +1,104 @@
+
+
+class AutomaticValueChecker(ValueChecker):
+    """Checks values automatically using PCC, all_close or rogerstanimoto based on type of the tensor.
+    if tensor is scalar all_close is used, if tensor is boolean based rogerstanimoto is used, else PCC is used."""
+
+    def __init__(
+        self, pcc: float = 0.99, rtol: float = 1e-05, atol: float = 1e-08, dissimilarity_threshold: float = 1e-03
+    ):
+        super().__init__(rtol, atol)
+        self.pcc = pcc
+        self.dissimilarity_threshold = (
+            dissimilarity_threshold  # threshold picked empirically. We will update it as TTNN evolves
+        )
+
+    def check(self, fw_out, co_out):
+        if not compare_with_golden(fw_out, co_out, self.pcc, self.rtol, self.atol, self.dissimilarity_threshold):
+            raise ValueError(
+                f"Data mismatch -> AutomaticValueChecker (compare_with_golden): framework_model={fw_out}, compiled_model={co_out}"
+            )
+
+
+class AllCloseValueChecker(ValueChecker):
+    """Checks values using torch.all_close."""
+
+    def check(self, fw_out, co_out):
+        assert fw_out.dtype not in [
+            torch.int32,
+            torch.int64,
+            torch.bool,
+        ], f"AllCloseValueChecker (all_close): all_close doesn't make sense for integer/bool types"
+
+        if not torch.allclose(fw_out, co_out, rtol=self.rtol, atol=self.atol):
+            atol, rtol = compute_required_tolerances(fw_out, co_out)
+            raise ValueError(
+                f"Data mismatch -> AllCloseValueChecker (all_close):\n"
+                f"- Current tolerances: rtol={self.rtol}, atol={self.atol}\n"
+                f"- Required tolerances for test to pass: rtol={rtol}, atol={atol}\n"
+                f"- Framework output: ({fw_out.shape})\n"
+                f"{fw_out}\n"
+                f"- Compiled model output: ({co_out.shape})\n"
+                f"{co_out}"
+            )
+
+
+class FullChecker(AutomaticValueChecker):
+    """Performs all checks including PCC and all_close."""
+
+    def __init__(
+        self, pcc: float = 0.99, rtol: float = 1e-05, atol: float = 1e-08, dissimilarity_threshold: float = 1e-03
+    ):
+        super().__init__(rtol, atol)
+        self.pcc = pcc
+        self.dissimilarity_threshold = (
+            dissimilarity_threshold  # threshold picked empirically. We will update it as TTNN evolves
+        )
+
+    def check(self, fw_out, co_out):
+        # Combines all checks (PCC and allclose)
+
+        if not compare_with_golden(fw_out, co_out, self.pcc, self.rtol, self.atol, self.dissimilarity_threshold):
+            raise ValueError(
+                f"Data mismatch -> FullChecker (compare_with_golden): framework_model={fw_out}, compiled_model={co_out}"
+            )
+
+        all_close_check = True
+        if fw_out.dtype not in [
+            torch.int32,
+            torch.int64,
+            torch.bool,
+        ]:  # allclose doesn't make sense for integer/bool types
```

Comment:
we should have an assertion if the tensor data type is bool/int

Reply:
I didn't put assertion on purpose, because most of the time people would want to call `FullChecker ` without thinking about dtypes or anything, so for that purpose I didn't put assertion here, but in `AllCloseValueChecker` where the whole purpose of the checker is to use `allclose` function I put the assertion regarding dtypes.

Reply:
Yeah, my bad... Maybe update the description of `FullChecker`:

```python
class FullChecker():
    """ Performs PCC and all_close for float tensors. """
```